### PR TITLE
Naboka Tymur IM-33

### PR DESCRIPTION
**Pull request by Homa4**

### CODE REVIEW

**Comment by alomj** [comment]
You have done a great job! This code is so clean!

**Comment by nikama-s** [comment]
I have reviewed all commits and can say, that I agree with everything

File: Tasks/filter.js [hunk 1]
```diff
@@ -1,15 +1,19 @@
...
-Filter = (T, t) => {
-  remove = [];
-  for (C of T) {
-    x = T.indexOf(C);
-    if (typeof T[x] !== t) {
-      remove.unshift(x);
+//1.change function name , change variables name variables name
+//2.change plice of const index now it doesn`t create on each iteraton of loop
+const filterArrayByTypeName = (array, typeName) => {
+  let removeElem = [];
```
Comment (tshemsedinov) [inline Tasks/filter.js:5]: const

File: Tasks/intersection.js [hunk 2]
```diff
@@ -1,25 +1,16 @@
...
+//3.change logic , delete else and change if condition and execution unit
+function intersectionOfTwoDictionaries(firstObject,secondObject) {
+  const keys = Object.keys(firstObject);
 
-intersection = function intersection(object_1, object_2) {
-  first_keys = Object.keys(object_1);
-  {
+  for (const elem of keys) {
+    if (firstObject[elem] != secondObject[elem]) {
+      delete firstObject[elem];
```
Comment (tshemsedinov) [inline Tasks/intersection.js:10]: Implement without delete

File: Tasks/intersection.js [hunk 3]
```diff
@@ -1,25 +1,16 @@
...
-      {
-      }
-    }
-    {
-    }
-  }
-  {
-  }
-  return object_1;
+  return firstObject;
```
Comment (tshemsedinov) [inline Tasks/intersection.js:13]: Do not mutate incoming arguments

File: Tasks/ip.js [hunk 4]
```diff
@@ -1,19 +1,20 @@
...
-      a[j] = parseInt(b);
-      if (isNaN(a[j])) return;
-      j++;
+//1.change name of function and variables
+//2.change type of loop from for..of to for . So (let j) we don`t need animore
+const parseip = (str) => {
+  let arr = [];
+  if (str === ''){
+    return;
+  } else {
```
Comment (tshemsedinov) [inline Tasks/ip.js:8]: Do not need else

File: Tasks/month.js [hunk 5]
```diff
@@ -1,26 +1,14 @@
...
-  'nov',
-  'dec',
-];
-
-Month = (s) => {
-  l = Months.length;
-  for (let i = 0; i < l; i++) {
-    if (s.toLowerCase().startsWith(Months[i])) return i + 1;
+const getMonthNumber = (monthNameStr) => {
+  const arrMonths = ['jan','feb','mar','apr','may','jun','jul','aug','sep','oct','nov','dec'];
```
Comment (tshemsedinov) [inline Tasks/month.js:7]: Move out of the function

File: Tasks/month.js [hunk 6]
```diff
@@ -1,26 +1,14 @@
...
+  for (let i = 0; i < arrMonths.length; i++) {
+    if (monthNameStr.toLowerCase().startsWith(arrMonths[i])) return i + 1;
```
Comment (tshemsedinov) [inline Tasks/month.js:9]: Long expression, decompose and move invariants out of the loop

File: Tasks/sum.js [hunk 7]
```diff
@@ -1,20 +1,14 @@
...
-      }
-    }
-  }
-  sum[0];
-  return sum[sum.length - 1];
-  sum.push(sum.length - 1);
-});
+const counterOfArrayNumbers = (arr) => {
+  let sumOfNumbers = arr.filter(elem => typeof elem === 'number').reduce((accum, elem) => accum + elem, 0);
+  return sumOfNumbers;
```
Comment (tshemsedinov) [inline Tasks/sum.js:10]: const

Comment (tshemsedinov) [reply Tasks/sum.js:10]: code allignment

File: Tasks/zip.js [hunk 8]
```diff
@@ -1,18 +1,22 @@
...
+      arr2[i] = arrCell;
+    }
+
+    if (arrCell[0] == undefined){
+      arr2.length -= 1
     }
-    if (CELL[0] == undefined) b.length -= 1
   }
-  return b;
+  return arr2;
```
Comment (tshemsedinov) [inline Tasks/zip.js:19]: Do not mutate incoming argumenta

File: Tasks/filter.js (hunk 1 above)
Comment (Homa4) [reply Tasks/filter.js]: code fixed

File: Tasks/intersection.js (hunk 2 above)
Comment (Homa4) [reply Tasks/intersection.js]: code fixed

File: Tasks/intersection.js (hunk 3 above)
Comment (Homa4) [reply Tasks/intersection.js]: code fixed

File: Tasks/ip.js (hunk 4 above)
Comment (Homa4) [reply Tasks/ip.js]: else was deleted

File: Tasks/month.js (hunk 5 above)
Comment (Homa4) [reply Tasks/month.js]: array was moved out of the function

File: Tasks/zip.js (hunk 8 above)
Comment (Homa4) [reply Tasks/zip.js]: fixed code

**Pull request closed without merging**